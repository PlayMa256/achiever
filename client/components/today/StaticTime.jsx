import React from 'react';
import PropTypes from 'prop-types';

import '../../styles/staticTime.styl';

const formatTime = value => (
	String(value).length === 1 ? `0${String(value)}` : String(value)
);

const displayTime = value => (
	value ? formatTime(value) : formatTime(0)
);

const StaticTime = ({ time, label, emphasis }) => (
	<div className={`static-time ${emphasis ? 'emphasis' : ''}`}>
		<label>
			{label}
		</label>
		<span className="time">
			{displayTime(time.hours)}:{displayTime(time.minutes)}
		</span>
	</div>
);

StaticTime.propTypes = {
	time: PropTypes.object,
	label: PropTypes.string.isRequired,
	emphasis: PropTypes.bool.isRequired
};

StaticTime.defaultProps = {
	time: { hours: 0, minutes: 0 }
};

export default StaticTime;
