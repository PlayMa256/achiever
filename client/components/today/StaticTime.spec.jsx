import React from 'react';
import { shallow } from 'enzyme';

import StaticTime from './StaticTime';

describe('StaticTime', () => {
	describe('render', () => {
		it('should render properly with minimum props', () => {
			const wrapper = shallow(<StaticTime label="entry" emphasis />);
			expect(wrapper).toMatchSnapshot();
		});
	})
	describe('display time', () => {
		it('should display time if anything comes', () => {
			const propTime = { hours: "", minutes: 10};
			const wrapper = shallow(<StaticTime
				time={propTime}
				label="entry"
				emphasis
			/>);
			expect(wrapper).toMatchSnapshot();
		})
	});
});

