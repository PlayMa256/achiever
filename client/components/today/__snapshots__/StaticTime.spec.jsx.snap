// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticTime display time should display time if anything comes 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StaticTime
    emphasis={false}
    label="entry"
    time={
      Object {
        "hours": "",
        "minutes": 10,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          entry
        </label>,
        <span
          className="time"
        >
          00
          :
          10
        </span>,
      ],
      "className": "static-time ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "entry",
        },
        "ref": null,
        "rendered": "entry",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "00",
            ":",
            "10",
          ],
          "className": "time",
        },
        "ref": null,
        "rendered": Array [
          "00",
          ":",
          "10",
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            entry
          </label>,
          <span
            className="time"
          >
            00
            :
            10
          </span>,
        ],
        "className": "static-time ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "entry",
          },
          "ref": null,
          "rendered": "entry",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "00",
              ":",
              "10",
            ],
            "className": "time",
          },
          "ref": null,
          "rendered": Array [
            "00",
            ":",
            "10",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`StaticTime render should render properly with minimum props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StaticTime
    emphasis={true}
    label="entry"
    time={
      Object {
        "hours": 0,
        "minutes": 0,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
          entry
        </label>,
        <span
          className="time"
        >
          00
          :
          00
        </span>,
      ],
      "className": "static-time emphasis",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "entry",
        },
        "ref": null,
        "rendered": "entry",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "00",
            ":",
            "00",
          ],
          "className": "time",
        },
        "ref": null,
        "rendered": Array [
          "00",
          ":",
          "00",
        ],
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            entry
          </label>,
          <span
            className="time"
          >
            00
            :
            00
          </span>,
        ],
        "className": "static-time emphasis",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "entry",
          },
          "ref": null,
          "rendered": "entry",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "00",
              ":",
              "00",
            ],
            "className": "time",
          },
          "ref": null,
          "rendered": Array [
            "00",
            ":",
            "00",
          ],
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
